packages/terminal-external/src/electron-browser/terminal-external-contribution.ts

```diff
@@ -0,0 +1,128 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import {
+    Command,
+    CommandContribution,
+    CommandRegistry
+} from '@theia/core/lib/common';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import { QuickPickService } from '@theia/core/lib/common/quick-pick-service';
+import {
+    KeybindingContribution,
+    KeybindingRegistry,
+    LabelProvider
+} from '@theia/core/lib/browser';
+import { EditorManager } from '@theia/editor/lib/browser/editor-manager';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { TerminalExternalService } from '../common/terminal-external';
+import { TerminalExternalPreferenceService } from './terminal-external-preference';
+
+export namespace TerminalExternalCommands {
+    export const OPEN_NATIVE_CONSOLE: Command = {
+        id: 'workbench.action.terminal.openNativeConsole',
+        label: 'Open New External Terminal'
+    };
+}
+
+@injectable()
+export class TerminalExternalFrontendContribution implements CommandContribution, KeybindingContribution {
+
+    @inject(EditorManager)
+    private readonly editorManager: EditorManager;
+
+    @inject(EnvVariablesServer)
+    private readonly envVariablesServer: EnvVariablesServer;
+
+    @inject(LabelProvider)
+    private readonly labelProvider: LabelProvider;
+
+    @inject(QuickPickService)
+    private readonly quickPickService: QuickPickService;
+
+    @inject(TerminalExternalService)
+    private readonly terminalExternalService: TerminalExternalService;
+
+    @inject(TerminalExternalPreferenceService)
+    private readonly terminalExternalPreferences: TerminalExternalPreferenceService;
+
+    @inject(WorkspaceService)
+    private readonly workspaceService: WorkspaceService;
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(TerminalExternalCommands.OPEN_NATIVE_CONSOLE, {
+            execute: () => this.openTerminalExternal()
+        });
+    }
```

Comment:
Should we make it `async` (for callers)?

```suggestion
    registerCommands(commands: CommandRegistry): void {
        commands.registerCommand(TerminalExternalCommands.OPEN_NATIVE_CONSOLE, {
            execute: async () => await this.openTerminalExternal()
        });
    }
```